src/sentry/seer/similarity/utils.py

```diff
@@ -197,59 +205,15 @@ def get_stacktrace_string(data: dict[str, Any], platform: str | None = None) ->
     if exceptions and exceptions[0].get("id") == "chained-exception":
         exceptions = exceptions[0].get("values")
 
-    frame_count = 0
-    html_frame_count = 0  # for a temporary metric
-    is_frames_truncated = False
-    has_no_filename_or_module = False
-    stacktrace_str = ""
-    found_non_snipped_context_line = False
-
     metrics.distribution("seer.grouping.exceptions.length", len(exceptions))
 
-    def _process_frames(frames: list[dict[str, Any]]) -> list[str]:
```

Comment:
Now that we use a dictionary to track the metrics it is not as ugly to take this function out of here.